csds-client/client/v2/client_test.go

```diff
@@ -162,11 +162,98 @@ func TestVisualization(t *testing.T) {
 	if err != nil {
 		t.Errorf("Read From File Failure: %v", err)
 	}
-	if err := clientutil.Visualize(responsejson, false); err != nil {
+	if err := clientUtil.Visualize(responsejson, false); err != nil {
 		t.Errorf("Visualization Failure: %v", err)
 	}
 	want := "digraph G {\nrankdir=LR;\n\\\"test_lds_0\\\"->\\\"test_rds_0\\\"[ arrowsize=0.3, penwidth=0.3 ];\n\\\"test_lds_0\\\"->\\\"test_rds_1\\\"[ arrowsize=0.3, penwidth=0.3 ];\n\\\"test_rds_0\\\"->\\\"test_cds_0\\\"[ arrowsize=0.3, penwidth=0.3 ];\n\\\"test_rds_0\\\"->\\\"test_cds_1\\\"[ arrowsize=0.3, penwidth=0.3 ];\n\\\"test_rds_1\\\"->\\\"test_cds_1\\\"[ arrowsize=0.3, penwidth=0.3 ];\n\\\"test_cds_0\\\" [ color=\\\"#34A853\\\", fillcolor=\\\"#34A853\\\", fontcolor=white, fontname=Roboto, label=CDS0, shape=box, style=\\\"\"filled,rounded\"\\\" ];\n\\\"test_cds_1\\\" [ color=\\\"#34A853\\\", fillcolor=\\\"#34A853\\\", fontcolor=white, fontname=Roboto, label=CDS1, shape=box, style=\\\"\"filled,rounded\"\\\" ];\n\\\"test_lds_0\\\" [ color=\\\"#4285F4\\\", fillcolor=\\\"#4285F4\\\", fontcolor=white, fontname=Roboto, label=LDS0, shape=box, style=\\\"\"filled,rounded\"\\\" ];\n\\\"test_rds_0\\\" [ color=\\\"#FBBC04\\\", fillcolor=\\\"#FBBC04\\\", fontcolor=white, fontname=Roboto, label=RDS0, shape=box, style=\\\"\"filled,rounded\"\\\" ];\n\\\"test_rds_1\\\" [ color=\\\"#FBBC04\\\", fillcolor=\\\"#FBBC04\\\", fontcolor=white, fontname=Roboto, label=RDS1, shape=box, style=\\\"\"filled,rounded\"\\\" ];\n\n}\n"
-	if err := clientutil.OpenBrowser("http://dreampuf.github.io/GraphvizOnline/#" + want); err != nil {
+	if err := clientUtil.OpenBrowser("http://dreampuf.github.io/GraphvizOnline/#" + want); err != nil {
 		t.Errorf("Open want graph failure: %v", err)
 	}
 }
+
+// TestNodeIdPrefixFilter tests node_id prefix filter
+func TestNodeIdPrefixFilter(t *testing.T) {
+	c := ClientV2{
+		opts: client.ClientOptions{
+			Platform:      "gcp",
+			FilterMode:    "prefix",
+			FilterPattern: "test",
+		},
+	}
+	filename, _ := filepath.Abs("./response_for_filter.json")
+	responsejson, err := ioutil.ReadFile(filename)
+	if err != nil {
+		t.Errorf("Read From File Failure: %v", err)
+	}
+	var response csdspb_v2.ClientStatusResponse
+	if err = protojson.Unmarshal(responsejson, &response); err != nil {
+		t.Errorf("Read From File Failure: %v", err)
+	}
+	out := clientUtil.CaptureOutput(func() {
+		if err := printOutResponse(&response, c.opts); err != nil {
+			t.Errorf("Print out response error: %v", err)
+		}
+	})
+	want := "Client ID                                          xDS stream type                Config Status                  \ntest_node_1                                        test_stream_type1              N/A                            \ntest_node_2                                        test_stream_type2              N/A                            \ntest_node_3                                        test_stream_type3              N/A                            \n"
```

Comment:
can this be broken down to multiple lines? same elsewhere.

Reply:
resolved